build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/extractor/GoRun.java

```diff
@@ -0,0 +1,218 @@
+    private CommandExecutor goCommandExecutor;
+    private String goCmdArgs;
+    private String resolutionRepository;
+    private String resolverUsername;
+    private String resolverPassword;
+    private Map<String, String> env;
+
+    /**
+     * Run go command and collect dependencies.
+     *
+     * @param goCmdArgs     - Go cmd args.
+     * @param path          - Path to directory contains go.mod.
+     * @param clientBuilder - Client builder for resolution.
+     * @param repo          - Artifactory's repository for resolution.
+     * @param username      - Artifactory's username for resolution.
+     * @param password      - Artifactory's password for resolution.
+     * @param logger        - The logger.
+     * @param env           - Environment variables to use during npm execution.
+     */
+    public GoRun(String goCmdArgs, Path path, ArtifactoryBuildInfoClientBuilder clientBuilder, String repo, String username, String password, Log logger, Map<String, String> env) throws IOException {
+        super(clientBuilder, path, logger);
+        this.goCmdArgs = goCmdArgs;
+        this.resolutionRepository = repo;
+        this.resolverUsername = username;
+        this.resolverPassword = password;
+        this.env = env;
+    }
+
+    public Build execute() {
+        try (ArtifactoryBuildInfoClient artifactoryClient = (clientBuilder != null ? (ArtifactoryBuildInfoClient) clientBuilder.build() : null)) {
+            if (artifactoryClient != null) {
+                preparePrerequisites(resolutionRepository, artifactoryClient);
+                setResolverAsGoProxy(artifactoryClient);
+            }
+            this.goCommandExecutor = new CommandExecutor(GO_CLIENT_CMD, env);
+            runCmd();
+            collectDependencies();
+            return createBuild(null, dependenciesList);
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    /*
+     * In order to use Artifactory as a resolver we need to set GOPROXY env var with Artifactory details.
+     * Wa also support fallback to VCS in case pkg doesn't exist in Artifactort,
+     */
+    private void setResolverAsGoProxy(ArtifactoryBuildInfoClient client) throws Exception {
+        URL rtUrl = new URL(client.getArtifactoryUrl());
+        URIBuilder proxyUrlBuilder = new URIBuilder()
+                .setScheme(rtUrl.getProtocol())
+                .setUserInfo(resolverUsername, resolverPassword)
+                .setHost(rtUrl.getHost())
+                .setPath(rtUrl.getPath() + ARTIFACTORY_GO_API + resolutionRepository);
+        String proxyValue = proxyUrlBuilder.build().toURL().toString() + "," + GOPROXY_VCS_FALLBACK;
+        env.put(GOPROXY_ENV_VAR, proxyValue);
+    }
+
+    private void runCmd() throws Exception {
+        List<String> args = new ArrayList<>();
+        for (String arg : goCmdArgs.split(" ")) {
+            args.add(arg);
+        }
+        runGoCmd(args);
+    }
```

Comment:
I think this function is redundant, because we can do its functionality in one line inside 'runGoGmd':
List<String> args = Lists.newArrayList(goCmdArgs.split(" "));

Reply:
Eyal told me we should try to avoid using guava as much as possible in build-info code. Is there other one-line we can use?

Reply:
List<String> args = new ArrayList<>(Arrays.asList(goCmdArgs.split(" ")));